Framework/DataHandling/src/ProcessBankData.cpp

```diff
@@ -33,6 +33,33 @@ ProcessBankData::ProcessBankData(
   m_cost = static_cast<double>(numEvents);
 }
 
+namespace {
+// this assumes that last_pulse_index is already to the point of including this
+// one so we only need to search forward
+inline size_t
+getPulseIndex(const size_t event_index, const size_t last_pulse_index,
+              const boost::shared_ptr<std::vector<uint64_t>> &event_index_vec) {
```

Comment:
consider just passing `const std::vector<uint64_t> &event_index_vec`

Reply:
Is that faster?

Reply:
It makes the interface cleaner. I pointer may be NULL but a reference cannot be.

In principle, you should have a check for NULL in this function, while passing a reference moves that responsibility to the calling code.